src/lib/storage.ts

```diff
@@ -121,6 +127,31 @@ export class Storage implements PeprStore {
     });
   };
 
+  /**
+   * Creates a promise and subscribes to the store, the promise resolves when
+   * the key is removed from the store.
+   *
+   * @param key - The key to add into the store
+   * @returns
+   */
+  removeItemAndWait = (key: string) => {
+    this.#dispatchUpdate("remove", [key]);
+    return new Promise<void>((resolve, reject) => {
+      const unsubscribe = this.subscribe(data => {
+        if (!data[key]) {
```

Comment:
```suggestion
        if (!data.hasOwnProperty(key)) {
```

Something like this would probably be better because `!data[key]` would return true anytime the content was falsy (0, false, null, undefined, etc.).

Using `.hasOwnProperty()` makes sure that there _is no property_ for the key that was removed and so won't get tripped up if someone tries to actually store a falsy value.